lua/telescope/builtin/diagnostics.lua

```diff
@@ -0,0 +1,142 @@
+local conf = require("telescope.config").values
+local finders = require "telescope.finders"
+local make_entry = require "telescope.make_entry"
+local pickers = require "telescope.pickers"
+
+local diagnostics = {}
+
+local convert_diagnostic_type = function(severities, severity)
+  -- convert from string to int
+  if type(severity) == "string" then
+    -- make sure that e.g. error is uppercased to ERROR
+    return severities[severity:upper()]
+  end
+  -- otherwise keep original value, incl. nil
+  return severity
+end
+
+local filter_diag_severity = function(opts, severity)
```

Comment:
`vim.diagnostic` now supports this sort of functionality OOTB via `:h vim.diagnostic.severity`, which we could (but ofc don't have to) follow.

Reply:
I thought it can only do exact. That is great. For now i wouldnt change our interface just forward it into `get`